actors/builtin/miner/miner_state.go

```diff
@@ -965,25 +989,24 @@ func (st *State) AddLockedFunds(store adt.Store, currEpoch abi.ChainEpoch, vesti
 		vestThisTime := big.Sub(targetVest, vestedSoFar)
 		vestedSoFar = targetVest
 
-		// Load existing entry, else set a new one
-		key := EpochKey(vestEpoch)
-		lockedFundEntry := big.Zero()
-		_, err = vestingFunds.Get(key, &lockedFundEntry)
-		if err != nil {
-			return err
-		}
-
-		lockedFundEntry = big.Add(lockedFundEntry, vestThisTime)
-		err = vestingFunds.Set(key, &lockedFundEntry)
-		if err != nil {
-			return err
+		// epoch already exists. Load existing entry
+		// and update amount.
+		if index, ok := epochToIndex[vestEpoch]; ok {
+			currentAmt := vestingFunds.Funds[index].Amount
+			vestingFunds.Funds[index].Amount = big.Add(currentAmt, vestThisTime)
+		} else {
+			// create a new entry and insert at the correct position
```

Comment:
This comment is a bit misleading. Instead say just blindly appended and sorted later.